mm2src/mm2_main/src/lp_swap.rs

```diff
@@ -1003,7 +994,10 @@ pub async fn my_recent_swaps_rpc(ctx: MmArc, req: Json) -> Result<Response<Vec<u
     let mut swaps = Vec::with_capacity(db_result.uuids.len());
     for uuid in db_result.uuids.iter() {
         let swap_json = match SavedSwap::load_my_swap_from_db(&ctx, *uuid).await {
-            Ok(Some(swap)) => json::to_value(MySwapStatusResponse::from(&swap)).unwrap(),
+            Ok(Some(mut swap)) => {
+                swap.hide_secrets();
```

Comment:
What do you think about adding `SavedSwap::load_my_swap_from_db_without_secrets`? I believe it'll help prevent disclosure of the secrets.

Reply:
It might be unclear when to use `load_my_swap_from_db_without_secrets` or `load_my_swap_from_db`. I think it will be more reliable to hide secrets in `MySwapStatusResponse::from` instead.